ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/ForeignKeyTableTableJoinBuilderTest.java

```diff
@@ -145,10 +155,11 @@ public void shouldDoLeftJoinOnNonKey() {
     // When:
     final KTableHolder<Struct> result = join.build(planBuilder, planInfo);
 
+    final ExpressionEvaluator e = mock(ExpressionEvaluator.class);
     // Then:
     verify(leftKTable).leftJoin(
         same(rightKTable),
-        eq(new KsqlKeyExtractor<>(1)),
+        any(KsqlKeyExtractor.class),
```

Comment:
@vcrfxia After the refactoring to always use `ExpressionEvaluator`, it pretty hard to verify that `KsqlKeyExtractor` is created correctly now... I just simplified this test to make it pass for now. But this change kinda make the test useless... Thoughts?

Reply:
If we really want to test this, we can use an ArgumentCaptor to capture the KsqlKeyExtractor, add a getter for the expression evaluator (visible for testing only), get the expression from the expression evaluator, and verify that the expression is a column reference expression for the appropriate column. I don't think this is too much effort if we wrap it in a helper method.

As you pointed out in the first version of this PR, test coverage for this (and some of the other unit tests) is lacking at the moment since we only test single columns for the left join expression, and not any non-column expressions. If we want to extend test coverage to include non-column expressions, I think we'll have to do something similar anyway. (I'm happy deferring this to a follow-up PR if you prefer since I know you've got some time constraints.)

Reply:
Oh nice! #gold